### PR TITLE
Feature: Messaging

### LINKED ISSUE #76: Inbox and messages for invitations

### PR DESCRIPTION
**Pull request by amundlrohne**
Anonymized view of candidates on project page and dynamic updates. 

Notification system for employees. 

Done so far: 
- Accept and decline invitations. 
- Notify employee when citizen accepts or declines. 
- Fetch data based on candidature status (no contact information if candidate is not "ACCEPTED" or "COMPLETED".
- Add timestamp in addition to date on messages.
- Make separate views for invitations and messages.

   
Consider: 
- Adding "EXPIRED" status to candidature. (should be done when CRON jobs are implemented).

Messaging system

### CODE REVIEW

File: apps/api/src/services/CandidateService.ts [hunk 1]
```diff
@@ -24,22 +27,52 @@ export class CandidateService extends BaseService<Candidate> {
...
+            .leftJoinAndSelect('candidate.user', 'user')
+            .getMany();
+
+        const anonymizedCandidates = await this._candidateRepository
+            .createQueryBuilder('candidate')
+            .leftJoinAndSelect('candidate.insight', 'insight', 'insight.id = :id', { id })
+            // eslint-disable-next-line quotes
+            .where("(candidate.status != 'ACCEPTED' AND candidate.status != 'COMPLETED') AND insight.id = :id", { id })
+            .getMany();
+
```
Comment (perkynades) [inline apps/api/src/services/CandidateService.ts]: I think to simplify this method and make it more digestable these database queries should be separeted into private methods

File: apps/api/src/services/MessageService.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        return this.create(message);
+    }
+
+    async createDeclineCandidatureMessage(insightId: string | number): Promise<Message> {
+        const insightService = new InsightService(this._database);
+        const insight = await insightService.getById(insightId);
+
+        const message = messageTemplates.candidateDeclined(insight);
+        return this.create(message);
+    }
```
Comment (perkynades) [inline apps/api/src/services/MessageService.ts:57]: A general solution with regards to future new types of messages should be considered, a factory pattern might be relevant.

File: apps/frontend/src/components/messages/Message.module.scss [hunk 3]
```diff
@@ -0,0 +1,39 @@
+.buttonGroup {
+    display: flex;
+    flex-direction: row;
+    justify-content: right;
+    gap: 1ch;
```
Comment (perkynades) [inline apps/frontend/src/components/messages/Message.module.scss]: Use rem :)

File: apps/frontend/src/components/misc/panelTabs/PanelTabs.module.scss [hunk 4]
```diff
@@ -0,0 +1,36 @@
+.wrapper {
+    position: absolute;
+    right: calc(((100% - 60rem) / 2) + 60rem + 1rem);
```
Comment (perkynades) [inline apps/frontend/src/components/misc/panelTabs/PanelTabs.module.scss]: Dont know if it is possible in scss, but a variable to explain this magic calculation would be nice

**Review by perkynades — CHANGES_REQUESTED** [review]
Very good work, DittNAV should implement the looks of this. Take a look at my comment about using a factory pattern for creating the different messages :)

File: apps/api/src/services/MessageService.ts (hunk 2 above)
Comment (amundlrohne) [reply apps/api/src/services/MessageService.ts:57]: Resolved over conversation on slack.

File: apps/api/src/services/CandidateService.ts (hunk 1 above)
Comment (amundlrohne) [reply apps/api/src/services/CandidateService.ts]: Done!

File: apps/frontend/src/components/messages/Message.module.scss (hunk 3 above)
Comment (amundlrohne) [reply apps/frontend/src/components/messages/Message.module.scss]: Done!

File: apps/frontend/src/components/misc/panelTabs/PanelTabs.module.scss (hunk 4 above)
Comment (amundlrohne) [reply apps/frontend/src/components/misc/panelTabs/PanelTabs.module.scss]: Done!

**Review by perkynades — APPROVED** [review]

**Pull request merged**